pkg/tsdb/prometheus/client/client.go

```diff
@@ -77,36 +78,51 @@ func (c *Client) QueryExemplars(ctx context.Context, q *models.Query) (*http.Res
 	qs.Set("start", formatTime(tr.Start))
 	qs.Set("end", formatTime(tr.End))
 
-	return c.fetch(ctx, c.method, u, qs)
+	return c.fetch(ctx, c.method, u, qs, nil)
 }
 
-func (c *Client) QueryResource(ctx context.Context, method string, p string, qs url.Values) (*http.Response, error) {
-	u, err := url.ParseRequestURI(c.baseUrl)
+type FetchReq struct {
+	Method      string
+	Url         *url.URL
+	QueryString url.Values
+}
+
+func (c *Client) QueryResource(ctx context.Context, req *backend.CallResourceRequest) (*http.Response, error) {
+	// The way URL is represented in CallResourceRequest and what we need for the fetch function is different
+	// so here we have to do a bit of parsing, so we can then compose it with the base url in correct way.
+	baseUrlParsed, err := url.ParseRequestURI(c.baseUrl)
+	if err != nil {
+		return nil, err
+	}
+	reqUrlParsed, err := url.Parse(req.URL)
 	if err != nil {
 		return nil, err
 	}
 
-	u.Path = path.Join(u.Path, p)
+	baseUrlParsed.Path = path.Join(baseUrlParsed.Path, req.Path)
+	baseUrlParsed.RawQuery = reqUrlParsed.RawQuery
 
-	return c.fetch(ctx, method, u, qs)
+	return c.fetch(ctx, req.Method, baseUrlParsed, nil, req.Body)
 }
 
-func (c *Client) fetch(ctx context.Context, method string, u *url.URL, qs url.Values) (*http.Response, error) {
-	if strings.ToUpper(method) == http.MethodGet {
+func (c *Client) fetch(ctx context.Context, method string, u *url.URL, qs url.Values, body []byte) (*http.Response, error) {
+	// The qs arg seems to be used in some callers of this method, but you can already pass them in the URL object
+	if strings.ToUpper(method) == http.MethodGet && qs != nil {
```

Comment:
Don't think so, it's the way it is because it is used for streaming and that probably needs to be refactored a bit and we probably not use it for resource calls right now as we mix different clients. Again something for refactor just didn't want to go into it right now.